tools/rosbag/src/player.cpp

```diff
@@ -278,6 +284,20 @@ void Player::printTime()
     }
 }
 
+void Player::waitForSubscribers() const
+{
+    bool all_topics_subscribed = false;
+    std::cout << "Waiting for subscribers." << std::endl;
+    while (!all_topics_subscribed) {
+        all_topics_subscribed = true;
+        typedef std::pair<std::string, ros::Publisher> value_type;
```

Comment:
Might be cleaner to `typedef std::map<std::string, ros::Publisher> PublisherMap` in the header, and then here you can use `PublisherMap::value_type` directly in the for loop without needing this duplicated definition.